filecoin-proofs/tests/api.rs

```diff
@@ -196,9 +196,167 @@ fn seal_lifecycle<Tree: 'static + MerkleTreeTrait>(
+fn test_seal_proof_aggregation_2_4kib_porep_id_v1_1_base_8() -> Result<()> {
+    let proofs_to_aggregate = 2;
+
+    let porep_id = ARBITRARY_POREP_ID_V1_1_0;
+    assert!(!is_legacy_porep_id(porep_id));
+    let verified = aggregate_proofs::<SectorShape4KiB>(
+        SECTOR_SIZE_4_KIB,
+        &porep_id,
+        ApiVersion::V1_1_0,
+        proofs_to_aggregate,
+    )?;
+    assert!(verified);
+
+    Ok(())
+}
+
+#[test]
+#[ignore]
+fn test_seal_proof_aggregation_1_32kib_porep_id_v1_1_base_8() -> Result<()> {
+    let proofs_to_aggregate = 1; // Requires auto-padding
+
+    let porep_id = ARBITRARY_POREP_ID_V1_1_0;
+    assert!(!is_legacy_porep_id(porep_id));
+    let verified = aggregate_proofs::<SectorShape32KiB>(
+        SECTOR_SIZE_32_KIB,
+        &porep_id,
+        ApiVersion::V1_1_0,
+        proofs_to_aggregate,
+    )?;
+    assert!(verified);
+
+    Ok(())
+}
+
+//#[test]
+//#[ignore]
+//fn test_seal_proof_aggregation_1024_2kib_porep_id_v1_1_base_8() -> Result<()> {
+//    let proofs_to_aggregate = 1024;
+//    inner_test_seal_proof_aggregation_2kib_porep_id_v1_1_base_8(proofs_to_aggregate)
+//}
+//
+//#[test]
+//#[ignore]
+//fn test_seal_proof_aggregation_65536_2kib_porep_id_v1_1_base_8() -> Result<()> {
+//    let proofs_to_aggregate = 65536;
+//    inner_test_seal_proof_aggregation_2kib_porep_id_v1_1_base_8(proofs_to_aggregate)
+//}
+
+fn inner_test_seal_proof_aggregation_2kib_porep_id_v1_1_base_8(
+    proofs_to_aggregate: usize,
+) -> Result<()> {
+    let porep_id_v1_1: u64 = 5; // This is a RegisteredSealProof value
+
+    let mut porep_id = [0u8; 32];
+    porep_id[..8].copy_from_slice(&porep_id_v1_1.to_le_bytes());
+    assert!(!is_legacy_porep_id(porep_id));
+
+    let rng = &mut XorShiftRng::from_seed(TEST_SEED);
+    let prover_fr: DefaultTreeDomain = Fr::random(rng).into();
+    let mut prover_id = [0u8; 32];
+    prover_id.copy_from_slice(AsRef::<[u8]>::as_ref(&prover_fr));
+
+    let mut commit_outputs = Vec::with_capacity(proofs_to_aggregate);
+    let mut commit_inputs = Vec::with_capacity(proofs_to_aggregate);
+
+    let (commit_output, commit_input) = create_seal_for_aggregation::<_, SectorShape2KiB>(
+        rng,
+        SECTOR_SIZE_2_KIB,
+        prover_id,
+        &porep_id,
+        ApiVersion::V1_1_0,
+    )?;
+
+    // duplicate a single proof to desired target for aggregation
+    for _ in 0..proofs_to_aggregate {
+        commit_outputs.push(commit_output.clone());
+        commit_inputs.extend(commit_input.clone());
+    }
+
+    let config = porep_config(SECTOR_SIZE_2_KIB, porep_id, ApiVersion::V1_1_0);
+    let aggregate_proof =
+        aggregate_seal_commit_proofs::<SectorShape2KiB>(config, commit_outputs.as_slice())?;
+    let verified = verify_aggregate_seal_commit_proofs::<SectorShape2KiB>(
+        config,
+        proofs_to_aggregate,
+        aggregate_proof,
+        commit_inputs,
+    )?;
+    assert!(verified);
+
+    Ok(())
+}
+
+fn aggregate_proofs<Tree: 'static + MerkleTreeTrait>(
```

Comment:
I'd suggest we put all of the test functions and helpers in a 
```
#[cfg(test)]
mod tests {
}
```
so all tests related functions like this one are only compiled in tests ?

Reply:
I thought this wasn't needed since we already separated the code into a test module?

Reply:
> I thought this wasn't needed since we already separated the code into a test module?

correct, this is only needed when writing tests inside a source file, but not when in the `tests` module